REGRESSION (280643@main): [ iOS EWS ] 11x media* (layout-tests) are constant/flaky failures

eric-carlson opened the pull request:
#### 8d9d83d24cd5f053edb1d3c88d8995d67f4161ad
<pre>
REGRESSION (280643@main): [ iOS EWS ] 11x media* (layout-tests) are constant/flaky failures
<a href="https://bugs.webkit.org/show_bug.cgi?id=276499">https://bugs.webkit.org/show_bug.cgi?id=276499</a>
<a href="https://rdar.apple.com/131558048">rdar://131558048</a>

Reviewed by Jer Noble.

Don&apos;t create the text track representation element until we know it will be used.

* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/compositing/video/video-object-fit-expected.txt: Removed.
* LayoutTests/platform/ios/compositing/video/video-object-position-expected.txt:
* LayoutTests/platform/mac-ventura/compositing/video/video-object-fit-expected.txt: Removed.
* LayoutTests/platform/mac-ventura/compositing/video/video-object-position-expected.txt: Removed.
* LayoutTests/platform/mac-wk2/compositing/video/video-object-fit-expected.txt: Removed.
* LayoutTests/platform/mac-wk2/compositing/video/video-object-position-expected.txt: Removed.
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::setTextTrackRepresentataionBounds):
(WebCore::HTMLMediaElement::setRequiresTextTrackRepresentation):
* Source/WebCore/html/HTMLMediaElement.h:

Canonical link: <a href="https://commits.webkit.org/281082@main">https://commits.webkit.org/281082@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8b593e492f48bb809f55fd19252a17ba73163cb9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/58643 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37970 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/11128 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/62270 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/9087 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/60772 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/45607 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/9285 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/47458 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6471 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60674 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/35554 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50714 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/28310 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/32297 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/8015 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/8091 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/54250 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/8287 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63973 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/2556 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/8285 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54778 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/2565 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50740 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/54862 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12963 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/2148 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/33799 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/34885 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/35969 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/34630 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jernoble reviewed (approved):
Could you also revert https://commits.webkit.org/280871@main as part of this patch?

eric-carlson committed:
REGRESSION (280643@main): [ iOS EWS ] 11x media* (layout-tests) are constant/flaky failures
https://bugs.webkit.org/show_bug.cgi?id=276499
rdar://131558048

Reviewed by Jer Noble.

Don't create the text track representation element until we know it will be used.

* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/compositing/video/video-object-fit-expected.txt: Removed.
* LayoutTests/platform/ios/compositing/video/video-object-position-expected.txt:
* LayoutTests/platform/mac-ventura/compositing/video/video-object-fit-expected.txt: Removed.
* LayoutTests/platform/mac-ventura/compositing/video/video-object-position-expected.txt: Removed.
* LayoutTests/platform/mac-wk2/compositing/video/video-object-fit-expected.txt: Removed.
* LayoutTests/platform/mac-wk2/compositing/video/video-object-position-expected.txt: Removed.
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::setTextTrackRepresentataionBounds):
(WebCore::HTMLMediaElement::setRequiresTextTrackRepresentation):
* Source/WebCore/html/HTMLMediaElement.h:

Canonical link: https://commits.webkit.org/281082@main
